### PR TITLE
Github Actions migration

### PR DESCRIPTION
**Pull request by Waterdrips**
## Description
Big PR to move over to GH Actions for building and publishing OFC containers

This will build containers and stack.yml and publish new versions with a tag that matches a release tag.

We will need to switch the chart to be using the release tag and pull from ghcr.io rather than dockerhub next.

Closes #684 

## How Has This Been Tested?

Containers are built and publised on relese tag
They are built and the chart tests are run on commits and PRs

## How are existing users impacted? What migration steps/scripts do we need?
We will be moving to GHCR for hosting images so will need to update the chart once they are pushed and public.
We will then need to update the chart to use the release tag as the image versions as we are going to be pushing those image tags from now on, rather than keeping each container on its own version. 

This should make maintaining the chart and stack.yml easier going forward. It will however mean that users should be updating all of ofc components on each release rather than just the components that change. This should be easier once the helm chart goes into ofc-bootstrap as theres a flag being added to run only the chart install

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests

### CODE REVIEW

**Comment by Waterdrips** [comment]
https://github.com/openfaas/faas/issues/1585 part of this

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        with:
+          fetch-depth: 1
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Run CI
+        run: sh ./contrib/ci.sh
+
+  # Test building the non-cli Containers
```
Comment (alexellis) [inline .github/workflows/build.yml:47]: Non-cli containers?

File: .github/workflows/publish.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        with:
+          fetch-depth: 1
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Login to Docker Registry
+        uses: docker/login-action@v1
+        with:
+          username: ${{ github.repository_owner }}
```
Comment (alexellis) [inline .github/workflows/publish.yaml:28]: Wait, who is the owner when it's an org? Can you test and find out please?

Comment (alexellis) [reply .github/workflows/publish.yaml:28]: If you use DOCKER_USERNAME it will just be easier

Comment (Waterdrips) [reply .github/workflows/publish.yaml:28]: secrets.DOCKER_USERNAME will cause every occurrence of the username to be starred out, which is annoying.

**Review by alexellis — APPROVED** [review]
Approved

File: .github/workflows/publish.yaml (hunk 2 above)
Comment (alexellis) [reply .github/workflows/publish.yaml:28]: We have that everywhere else and I don't think that openfaas can push images? I can't get a PTA for openfaas the owner.

**Comment by alexellis** [comment]
This looks great - we are getting there!

I'm not seeing the gitlab stack file being built? How about the aws one too?

**Pull request merged**